compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1970,57 +1970,127 @@ setTransformStrategyRootConfig(mlir::FunctionOpInterface entryPointFn,
   return success();
 }
 
-/// Sets the lowering configuration for a generic op implementing a
-/// transposition to use CPUDoubleTilingExpert pipeline.
-static LogicalResult
-setTransposeLikeOpRootConfig(mlir::FunctionOpInterface entryPointFn,
-                             linalg::GenericOp genericOp,
-                             const LinalgOpInfo &linalgOpInfo,
-                             const TargetMLTransformInfo &targetMLTransInfo) {
-  assert(!getLoweringConfig(genericOp) &&
-         "expected lowering_config is not set");
-
-  LLVM_DEBUG(KD_DBGS() << "Setting transpose-like op root configuration\n");
-
-  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
-  if (!hasAVX2Feature(targetAttr) ||
-      !x86TransposeLoweringPrecondition(genericOp)) {
-    return failure();
-  }
+/// Utility to compute the transpose vector sizes for X86.
```

Comment:
Need more details about the documentation. Without looking into implementation, it is not clear that the vectors are empty when it fails. Also, how about returning a boolean value which indicates if it succeeds or not?

Reply:
I quite like the empty state of `sizes` indicating failure so I've kept that, but updated the docs to make it clear 👍